packages/babel-plugin-transform-es2015-modules-commonjs/src/index.js

```diff
@@ -6,8 +6,82 @@ import {
   ensureStatementsHoisted,
   wrapInterop,
 } from "babel-helper-module-transforms";
+import simplifyAccess from "babel-helper-simple-access";
+
+export default function({ types: t, template }) {
+  const moduleAssertion = template(`
+    (function(){
+      throw new Error("The CommonJS 'module' variable is not available in ES6 modules.");
+    })();
+  `);
+  const exportsAssertion = template(`
+    (function(){
+      throw new Error("The CommonJS 'exports' variable is not available in ES6 modules.");
```

Comment:
This is also the case for `__dirname`, `__filename`, `require` et al but we can't really ban those until node gets a standard `import.meta` shape

Reply:
Yeah I'd be up for doing this for those too, but we can always do it behind a separate flag. For now at least this PR allows us to get closer.